code/tests/test_public_APIs.brs

```diff
@@ -351,6 +351,76 @@ sub TC_APIs_setExperienceCloudId()
     UTF_AssertNotInvalid(event.timestamp)
 end sub
 
+' target: getExperienceCloudId()
+' @Test
+sub TC_APIs_getExperienceCloudId()
+    _internal_const = _adb_InternalConstants()
+    sdkInstance = AdobeAEPSDKInit()
+
+    context = {
+        content: "test"
+    }
+    callback_result = "testECID"
+
+    sdkInstance.getExperienceCloudId(sub(ctx, result)
+        UTF_assertEqual({
+            content: "test"
+        }, ctx)
+        UTF_assertEqual(result, "testECID")
+    end sub, context)
+
+    event = GetGlobalAA()._adb_main_task_node["requestEvent"]
+    callbackInfo = sdkInstance._private.cachedCallbackInfo[event.uuid]
+
+    UTF_assertEqual(event.apiName, _internal_const.PUBLIC_API.GET_EXPERIENCE_CLOUD_ID)
+    UTF_assertInvalid(event.data)
+    UTF_AssertNotInvalid(event.uuid)
+    UTF_AssertNotInvalid(event.timestamp)
+    UTF_assertEqual(callbackInfo.context, context)
+    UTF_AssertNotInvalid(callbackInfo.timestampInMillis)
+    callbackInfo.cb(context, callback_result)
+end sub
+
+' target: getExperienceCloudId()
+' @Test
+sub TC_APIs_getExperienceCloudId_callbackTimeout()
+    _internal_const = _adb_InternalConstants()
+    sdkInstance = AdobeAEPSDKInit()
+
+    context = {
+        content: "test"
+    }
+    callback_result = "testECID"
+
+    sdkInstance.getExperienceCloudId(sub(ctx, result)
+        throw "should not be called"
+    end sub, context)
+
+    sleep(5001)
+
+    event = GetGlobalAA()._adb_main_task_node["requestEvent"]
+    callbackInfo = sdkInstance._private.cachedCallbackInfo[event.uuid]
+
+    UTF_assertEqual(event.apiName, _internal_const.PUBLIC_API.GET_EXPERIENCE_CLOUD_ID)
+    UTF_assertInvalid(event.data)
+    UTF_AssertNotInvalid(event.uuid)
+    UTF_AssertNotInvalid(event.timestamp)
+    UTF_assertEqual(callbackInfo.context, context)
+    UTF_AssertNotInvalid(callbackInfo.timestampInMillis)
+
+    sleep(5001)
```

Comment:
Is this the behaviour across all our APIs? Don't we signal timeout?

Reply:
Yes, currently we don't error for timeout. If we want to do that, I think we will have to loop through all the callbacks and check for timeouts and error out and delete.